rfc/RFC_for_CI_roles_extraction.md

```diff
@@ -0,0 +1,42 @@
+# RFC for CI roles extraction
+### An idea to extract common roles for deploying controller from contrail-project-config to be used by Contrail-Windows and DevOps teams.
+
+Author: Katarzyna Rybacka
+
+---
+
+
+## Abstract
+
+Currently, controller5.0 is being deployed in Contrail-Windows CI using roles copied from `contrail-project-config` repository([Link](https://github.com/Juniper/contrail-project-config)).
+There is a need to make this code more generic so it could be used by few teams simultaneously.
+This document specifies the proposed solution to this issue and requests discussion and suggestions for
+improvements.
+
+
+## Introduction
+
+In CI, specifically in `contrail-windows-ci` repository ([Link](https://github.com/Juniper/contrail-windows-ci)) there are three roles `contrail-ansible-deployer`, `kolla-provision-dockers`
+and `yum-repos-prepare` used to deploy controller5.0. The fact that this code is copied from another repository
+could cause some issues e.g. need for backporting code, when any change occures in source code.
+
+
+## Proposal
+
+The idea is to extract generic code used by Contrail-Windows and DevOps teams to another repository
+and make these roles more parameterized. All action steps are proposed to be performed by members of Contrail-Windows team.
+
+
+![img](../pictures/rfc_for_ci_roles_extraction.png)
```

Comment:
This image hurts my eyes. :(

* Consider redoing it using a component UML diagram or something similar.
* Current/desired state summaries shouldn't be present on a diagram

Reply:
This diagram should show source code dependencies between components as well as repositories in which they would reside. This should let us see if it violates any package principles. https://en.wikipedia.org/wiki/Package_principles